### PR TITLE
Fix: Update annotation structure, put CAIS annotations under annotation["assetInfo"]

### LINKED ISSUE #72: Presubmit check should validate mapping annotations keys
https://github.com/abcxyz/pmap/blob/main/pkg/mapping/processors/assetinventory.go#L162-L182

We use these keys (from CAIS) in the annotations. If somebody manually provides an annotation that has the same key, it could get overwritten by the CAIS processor.
We need to validate that the mapping is not using such well-known annotation keys.
Even better, CAIS info probably should be fully under a dedicated annotation. E.g.

```
annotations:
  assetInfo:
    iamPolicies:
    labels: foo
```

cc: @capri-xiyue

### PR DESCRIPTION
**Pull request by sailorlqh**
fix #72

### CODE REVIEW

File: pkg/mapping/processors/assetinventory_test.go [hunk 1]
```diff
@@ -143,33 +143,45 @@ func TestProcessor_UpdatedProcess(t *testing.T) {
 					Provider: "gcp",
 					Name:     "//pubsub.googleapis.com/projects/test-project/topics/test-topic",
 				},
-				Contacts: &v1alpha1.Contacts{Email: []string{"pmap.gmail.com"}},
+				Contacts: &v1alpha1.Contacts{Email: []string{"pmap@gmail.com"}},
```
Comment (yolocs) [inline pkg/mapping/processors/assetinventory_test.go:146]: Can we globally change to `@example.com`

File: apis/v1alpha1/validation.go [hunk 2]
```diff
@@ -31,5 +31,8 @@ func ValidateResourceMapping(m *ResourceMapping) (vErr error) {
 	if m.Resource.Provider == "" {
 		vErr = errors.Join(vErr, fmt.Errorf("empty resource provider"))
 	}
+	if _, ok := m.Annotations.AsMap()["assetInfo"]; ok {
```
Comment (yolocs) [inline apis/v1alpha1/validation.go:34]: Probably in this package - Can we define the well-known keys as (exposed) consts?

**Review by yolocs — CHANGES_REQUESTED** [review]

File: apis/v1alpha1/validation.go [hunk 3]
```diff
@@ -21,6 +21,8 @@ import (
 	"net/mail"
 )
 
+const WellKnownKey = "assetInfo"
```
Comment (capri-xiyue) [inline apis/v1alpha1/validation.go:24]: nits: using 
```
const (
	WellKnownKey = "assetInfo"
)
```

File: apis/v1alpha1/validation_test.go [hunk 4]
```diff
@@ -38,7 +38,7 @@ func TestValidateResouceMapping(t *testing.T) {
 					Name:     "//pubsub.googleapis.com/projects/test-project/topics/test-topic",
 				},
 				Contacts: &Contacts{
-					Email: []string{"pmap.gmail.com"},
+					Email: []string{"pmap.example.com"},
```
Comment (capri-xiyue) [inline apis/v1alpha1/validation_test.go:41]: change to @

Comment (sailorlqh) [reply apis/v1alpha1/validation_test.go:41]: I think this is the invalid email test, we want an invalid email here?

Comment (capri-xiyue) [reply apis/v1alpha1/validation_test.go:41]: Sorry, I missed that(the previous code lines were hidden by default). Maybe change it to `invalid.example.com` to make it more obvious

Comment (sailorlqh) [reply apis/v1alpha1/validation_test.go]: Done.

File: apis/v1alpha1/validation.go (hunk 3 above)
Comment (capri-xiyue) [reply apis/v1alpha1/validation.go]: Did you change the variable naming?

**Review by capri-xiyue — APPROVED** [review]

File: apis/v1alpha1/validation.go [hunk 5]
```diff
@@ -21,6 +21,10 @@ import (
 	"net/mail"
 )
 
+const (
+	AssetInfoKey = "assetInfo"
```
Comment (yolocs) [inline apis/v1alpha1/validation.go:25]: Go doesn't have enum so the convention is to put "type" as the first part of the variable name, e.g. `AnnotationKeyAssetInfo`, `AnnotationKeyOtherStuff`

**Review by yolocs — CHANGES_REQUESTED** [review]

File: apis/v1alpha1/validation.go (hunk 5 above)
Comment (sailorlqh) [reply apis/v1alpha1/validation.go]: Thanks for these comments, I will keep them in-mind! Already updated.

File: apis/v1alpha1/validation.go [hunk 6]
```diff
@@ -21,6 +21,10 @@ import (
 	"net/mail"
 )
 
+const (
+	AnnotationKeyAssetInfo = "assetInfo"
```
Comment (yolocs) [inline apis/v1alpha1/validation.go:25]: Comment

**Review by yolocs — APPROVED** [review]

**Pull request merged**